File: projects/m1/044-reduce-measures/solutions/enricozoboli/solution_reduce_measures.py
def reduce(units, measures):
    measures = measures.lower()
    if measures == "cup" or measures == "cups":
        cup = units
        return print(f"{cup} cup/s.")
    elif measures == "tablespoons" or measures == "tablespoon":
        cup = int(units) // 16
        tablespoon = int(units) % 16
        return print(f"{cup} cup/s, {tablespoon} tablespoon/s.")
    elif measures == "teaspoons" or measures == "teaspoon":
        cup = (int(units) // 3) // 16
        tablespoon = (int(units) // 3) % 16
        teaspoon = int(units) % 3
        return print(f"{cup} cup/s, {tablespoon} tablespoon/s,"
        f" {teaspoon} teaspoon/s.")

if __name__ == "__main__":
    reduce( 59, "teaspoons")
    reduce( 6, "cups")
    reduce( 18, "tablespoon")
    reduce( 1289, "teaspoon")
    reduce( 1, "teaspoon")
    reduce( 15, "tablespoon")
    
   
    

File: projects/m1/044-reduce-measures/solutions/valentinaBazzacchi/solution-reduce-measures.js
// Main Program
let userNumber = prompt("Enter the number of units");
let testNumber = /^\d+$/;
// Verify user input for the number of units, if isn't correct --> Error Message + Exit the Program!
if (!testNumber.test(userNumber)) {
    alert("Invalid number of units!")
// Else read the unit of measure by the user and call the function
} else {
    let userUnit = prompt("Enter the unit of measure: cup, tablespoon or teaspoon");
    reduceMeasures(userNumber, userUnit);
}

function reduceMeasures(number, unit) {
    let cup, tbsp, tsp;

    if (unit == "cup" || unit ==  "cups") {
    cup = number;
    return alert(cup + " cup/s");

    } else if (unit == "tablespoon" || unit == "tablespoons")  {
    cup = Math.floor( number / 16);
    tbsp = number % 16;
    return alert(cup + " cup/s, " + tbsp + " tbsp");

    } else if (unit == "teaspoon" || unit == "teaspoons") {
    cup = Math.floor( number / 48),
    tsp = number % 48;
    tbsp = Math.floor( tsp / 3);
    tsp = tsp % 3;
    return alert(cup + " cup/s, " + tbsp + " tbsp, " + tsp + " tsp");
    // If the unit of measure entered by the user is not among those specified in the previous conditions --> Error Message!
    } else {
    alert ("Invalid unit of measure!");
    }

} // end function